src/crypto/OutgoingRoomKeyRequestManager.js

```diff
@@ -107,6 +110,79 @@ export default class OutgoingRoomKeyRequestManager {
         });
     }
 
+    /**
+     * Cancel room key requests, if any match the given details
+     *
+     * @param {module:crypto~RoomKeyRequestBody} requestBody
+     *
+     * @returns {Promise} resolves when the request has been updated in our
+     *    pending list.
+     */
+    cancelRoomKeyRequest(requestBody) {
+        return this._cryptoStore.getOutgoingRoomKeyRequest(
+            requestBody,
+        ).then((req) => {
+            if (!req) {
+                // no request was made for this key
+                return;
+            }
+
+            if (req.state === ROOM_KEY_REQUEST_STATES.UNSENT) {
+                // just delete it
+                console.log(
+                    'deleting unnecessary room key request for ' +
+                    stringifyRequestBody(requestBody),
+                );
+                return this._cryptoStore.deleteOutgoingRoomKeyRequest(
+                    req.requestId, ROOM_KEY_REQUEST_STATES.UNSENT,
+                );
+            }
+
+            if (req.state !== ROOM_KEY_REQUEST_STATES.SENT) {
+                // the only other option is cancellation_pending, in which case
+                // there is nothing else to do here
+            }
```

Comment:
I feel you may be missing something here? Surely you mean:
```js
if (req.state !== ROOM_KEY_REQUEST_STATES.SENT) {
    return;
}
```

As an aside: I dislike and would discourage the use of "handle 2 cases, only 1 left so don't bother to check" since this will break in weird ways if we end up adding more states, plus it's clearer meaning you needn't document what you mean for the last case:

```js
if (req.state === ROOM_KEY_REQUEST_STATES.CANCELLATION_PENDING) {
    return;
}
```
in a function called `cancelRoomKeyRequest` is nice and clear that we've already sent the cancellation and therefore don't need to resend.